Python/optimizer.c

```diff
@@ -161,11 +162,24 @@ _PyOptimizer_BackEdge(_PyInterpreterFrame *frame, _Py_CODEUNIT *src, _Py_CODEUNI
         if (err < 0) {
             return NULL;
         }
-        _PyFrame_SetStackPointer(frame, stack_pointer);
-        return frame;
+        goto jump_to_destination;
+    }
+    int index = get_index_for_executor(frame->f_code, src);
+    if (index < 0) {
+        /* Out of memory. Don't raise and assume that the
+         * error will show up elsewhere.
+         * This might confuse an optimizer as if assumes that
+         * it has already optimixed this point */
+        Py_DECREF(executor);
+        goto jump_to_destination;
     }
     insert_executor(frame->f_code, src, index, executor);
+    assert(frame->prev_instr = src);
```

Comment:
This is an assignment!

```suggestion
    assert(frame->prev_instr == src);
```